src/lib.rs

```diff
@@ -685,6 +685,33 @@ impl Uuid {
         Uuid { bytes: b }
     }
 
+    /// Creates a v4 Uuid from rng bytes (ref issue #170)
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use uuid::Uuid;
+    /// use uuid::UuidBytes;
+    ///
+    /// let bytes:UuidBytes = [70, 235, 208, 238, 14, 109, 67, 201, 185, 13, 204, 195, 90, 145, 63, 62];
+    ///
+    /// let uuid = Uuid::from_rng_bytes(bytes);
+    /// let uuid = uuid.hyphenated().to_string();
+    ///
+    /// let expected_uuid = String::from("46ebd0ee-0e6d-43c9-b90d-ccc35a913f3e");
+    ///
+    /// assert_eq!(expected_uuid, uuid);
+    /// ```
+    ///
+    #[cfg(feature = "v4")]
+    pub fn from_rng_bytes(b: [u8; 16]) -> Uuid {
+        let mut uuid = Uuid { bytes: [0; 16] };
+        copy_memory(&mut uuid.bytes, &b);
```

Comment:
We also need to call `set_variant` and `set_version` here since the random bytes probably won't have the right values:

```rust
let mut uuid = Uuid { bytes: [0; 16] };
copy_memory(&mut uuid.bytes, &b);

uuid.set_variant(UuidVariant::RFC4122);
uuid.set_version(UuidVersion::Random);

uuid
```